libs/langchain-mongodb/tests/integration_tests/test_retriever_selfquerying.py

```diff
@@ -0,0 +1,231 @@
+"""Tests MongoDBAtlasSelfQueryRetriever and MongoDBStructuredQueryTranslator."""
+
+import os
+from typing import Sequence, Union
+
+import pytest
+from langchain.chains.query_constructor.schema import AttributeInfo
+from langchain.retrievers.self_query.base import SelfQueryRetriever
+from langchain_core.documents import Document
+from langchain_openai import ChatOpenAI
+
+from langchain_mongodb import MongoDBAtlasVectorSearch, index
+from langchain_mongodb.retrievers import MongoDBAtlasSelfQueryRetriever
+
+from ..utils import PatchedMongoDBAtlasVectorSearch
+
+DB_NAME = "langchain_test_db"
```

Comment:
This should be imported from `utils` now for consistency.

Reply:
Merged upstream/main and updated.